### DISCUSSION TITLE
Schedule scripts hourly?

### DISCUSSION THREAD

**Discussion by ghost** [amidaware/tacticalrmm#1725] [Q&A] [answered]
I know scripts can be scheduled daily etc but I have no guarantee that endpoints will be switched on at the scheduled time. 

How to run scripts hourly with no end date instead?

**Comment by wh1te909** [comment]
Just use a script check instead of a task. Set the run interval to 3600 seconds.

**Reply by silversword411** [reply]
Not ATM, only tasks can save to custom fields.
Task every hour is possible

**Comment by ghost** [comment]
Perfect.

Can script check save output to a custom field?

**Comment by ghost** [comment]
> Not ATM, only tasks can save to custom fields. Task every hour is possible

So go to Tasks, set to run daily and under "Advanced" set "Repeat Task every" to 1h ?

What to set so "Task repeat duration"?

**Reply by dinger1986** [reply]
You can have a check that triggers a task and use that as a collector task.

So for example if you are checking for software have a check for that software or a file which is written at the end of the task being ran, that then triggers the task.

**Reply by ghost** [reply]
I understand that. It's the getting it to run every hour in scheduling that I'm asking about.

**Reply by dinger1986** [reply]
ah sorry, for duration you want to set it under the new occurrence, as in if you have set your script to run every day set the duration to 23 hours

**Reply by ghost** [reply]
Maybe I'm not making myself clear. I want to run a script once an hour, every hour.

What do I set here to do it (see screenshot)?

<img width="1063" alt="Screenshot 2024-01-04 at 09 11 22" src="https://github.com/amidaware/tacticalrmm/assets/98236582/6937c781-dde6-4a27-9789-26b4f35e203d">

**Reply by silversword411** [reply, marked as answer]
Looks right for every hour run to me

**Reply by silversword411** [reply]
Actually, I didn't check the Task repeat duration. That should be 1d. Like this

![2024-01-04_073411 - brave_Tactical_RMM_-_Brave](https://github.com/amidaware/tacticalrmm/assets/13395348/11042308-d2d4-4e43-b447-2287cd99fb51)

**Reply by silversword411** [reply]
...and I think dinger is right. You'll probably get a double-run on the hour with 1d....so 23h Task repeat duration is probably better :)
